### DISCUSSION TITLE
How to use WinAuth.

### DISCUSSION THREAD

**Discussion by JDCain** [justcoding121/Titanium-Web-Proxy#870] [Q&A] [answered]
I am attempting to run Titanium Proxy on a Windows box and have Titanium pass the NTLM AUTH for the running user when requests are passed though the server but this does not work either with WinAuth true or false. 

WinAuth = true: I get a windowed error that the authentication failed.
WinAuth = false: I get a basic AUTH prompt I can cancel and then the site loads.

**Comment by justcoding121** [comment, marked as answer]
NTLM/Negotiate can't be transparently MITM'd. With HTTPS decryption on, the proxy has to authenticate *as itself* (the process identity), not relay the browser's NTLM handshake.

Options:

**1. Let the proxy auth with the process user**
```csharp
proxyServer.EnableWinAuth = true;
// process must run as a user that has access to the site
// or supply credentials:
proxyServer.WinAuthCredentialsProvider = async e =>
    new WinAuthCredentials("DOMAIN", "user", "password");
```

**2. Don't decrypt those hosts** (browser does NTLM itself):
```csharp
explicitEndPoint.BeforeTunnelConnectRequest += async (sender, e) =>
{
    if (e.HttpClient.Request.RequestUri.Host.Contains("intranet"))
        e.DecryptSsl = false;
};
```

`EnableWinAuth = false` is why you see a basic auth prompt ? the 401 just passes through.

Also: WinAuth is Windows-only, and connection pooling can interfere with NTLM (connection-bound). If it still fails with option 1, try disabling reuse for that traffic or use option 2.

Closing ? reopen with the exact host/error if you're still stuck.
